### PR TITLE
Filter contact list based on interview date and time feature

### LINKED ISSUE #71: As a Computer Science job seeker, I can filter my interview contact list based on their interview times
so that I can look at relevant interviews more conveniently based on the interview time range

### PR DESCRIPTION
**Pull request by Lalelulilulela**
closes #71
Filters contact list based on interview dates and times
- Command Format: filter tt/ INTERVIEW_TIME_RANGE [MORE INTERVIEW_TIME_RANGE]...
  - filter tt/ before/INTERVIEW_TIME 
  - filter tt/ after/INTERVIEW_TIME
  - filter tt/ from/INTERVIEW_TIME-INTERVIEW_TIME

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FilterInterviewTimeCommand.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            return false;
+        }
+
+        FilterInterviewTimeCommand otherFilterInterviewTimeCommand = (FilterInterviewTimeCommand) other;
+        return predicate.equals(otherFilterInterviewTimeCommand.predicate);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
```
Comment (dabzpengu) [inline src/main/java/seedu/address/logic/commands/FilterInterviewTimeCommand.java:67]: nice use of stringbuilder!

**Comment by dabzpengu** [comment]
LGTM

File: src/main/java/seedu/address/logic/commands/FilterInterviewTimeCommand.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                .add("interviewTime", predicate)
+                .toString();
+    }
```
Comment (Dexter-Wong) [inline src/main/java/seedu/address/logic/commands/FilterInterviewTimeCommand.java:70]: Good code structuring

File: src/main/java/seedu/address/logic/parser/FilterInterviewTimeCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    private static List<List<InterviewTime>> handleAfterInterviewTime(ArgumentMultimap argMultimap)
+            throws ParseException {
+        List<String> afterInterviewTimes = argMultimap.getAllValues(PREFIX_AFTER_INTERVIEW_TIME);
+        List<List<InterviewTime>> interviewTimes = new ArrayList<>();
+        for (String afterInterviewTime : afterInterviewTimes) {
+            interviewTimes.add(Arrays.asList(ParserUtil.parseInterviewTime(afterInterviewTime), null));
+        }
+        return interviewTimes;
+    }
+
```
Comment (Dexter-Wong) [inline src/main/java/seedu/address/logic/parser/FilterInterviewTimeCommandParser.java:93]: Good use of argumentMultimap

File: src/main/java/seedu/address/model/person/InterviewTimeContainsKeywordsPredicate.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                (InterviewTimeContainsKeywordsPredicate) other;
+        return keywords.equals(otherSalaryContainsKeywordsPredicate.keywords);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("interviewTimes", keywords).toString();
+    }
+}
```
Comment (Dexter-Wong) [inline src/main/java/seedu/address/model/person/InterviewTimeContainsKeywordsPredicate.java:47]: Good logic flow

File: src/test/java/seedu/address/logic/commands/FilterInterviewTimeCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        InterviewTimeContainsKeywordsPredicate predicate =
+                new InterviewTimeContainsKeywordsPredicate(Arrays.asList(
+                        Arrays.asList(null, new InterviewTime("121220221430")),
+                        Arrays.asList(new InterviewTime("121220221430"), null)
+                ));
+        FilterInterviewTimeCommand filterInterviewTimeCommand = new FilterInterviewTimeCommand(predicate);
+        String expected = FilterInterviewTimeCommand.class.getCanonicalName() + "{interviewTime=" + predicate + "}";
+        assertEquals(expected, filterInterviewTimeCommand.toString());
+    }
+}
```
Comment (Dexter-Wong) [inline src/test/java/seedu/address/logic/commands/FilterInterviewTimeCommandTest.java:107]: Nice tests

File: src/test/java/seedu/address/model/person/InterviewTimeContainsKeywordsPredicateTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                Arrays.asList(new InterviewTime("121220270000"), null)
+        );
+        InterviewTimeContainsKeywordsPredicate predicate = new InterviewTimeContainsKeywordsPredicate(keywords);
+
+        String expected =
+                InterviewTimeContainsKeywordsPredicate.class.getCanonicalName()
+                        + "{interviewTimes=" + keywords + "}";
+        assertEquals(expected, predicate.toString());
+    }
+}
```
Comment (Dexter-Wong) [inline src/test/java/seedu/address/model/person/InterviewTimeContainsKeywordsPredicateTest.java:105]: Good test coverage

**Review by Dexter-Wong — COMMENTED** [review]
Lgtm

**Pull request merged**